app/controllers/physical_storage_controller.rb

```diff
@@ -4,12 +4,29 @@ class PhysicalStorageController < ApplicationController
   include Mixins::GenericSessionMixin
   include Mixins::MoreShowActions
   include Mixins::BreadcrumbsMixin
+  include Mixins::GenericButtonMixin
 
   before_action :check_privileges
   before_action :session_data
   after_action :cleanup_action
   after_action :set_session_data
 
+  def specific_buttons(pressed)
+    case pressed
+    when 'physical_storage_new'
+      javascript_redirect(:action => 'new')
+    else
+      return false
+    end
+    true
+  end
+
+  def new
```

Comment:
`new` as an instance method concerns me....is this a common pattern @h-kataria ?

Reply:
Yup
```
ag -c "def new" app/controllers/
app/controllers/cloud_network_controller.rb:1
app/controllers/cloud_object_store_container_controller.rb:1
app/controllers/floating_ip_controller.rb:1
app/controllers/ansible_credential_controller.rb:1
app/controllers/network_router_controller.rb:1
app/controllers/chargeback_rate_controller.rb:2
app/controllers/mixins/ems_common.rb:1
app/controllers/mixins/manager_controller_mixin.rb:1
app/controllers/cloud_volume_controller.rb:1
app/controllers/generic_object_definition_controller.rb:1
app/controllers/host_aggregate_controller.rb:1
app/controllers/cloud_subnet_controller.rb:2
app/controllers/security_group_controller.rb:1
app/controllers/flavor_controller.rb:1
app/controllers/auth_key_pair_cloud_controller.rb:1
app/controllers/ansible_repository_controller.rb:1
app/controllers/miq_ae_class_controller.rb:6
app/controllers/cloud_tenant_controller.rb:1

```

Reply:
https://guides.rubyonrails.org/action_controller_overview.html#methods-and-actions